### PR TITLE
Add used_by_otel field in modules.py and update update-go invoke task

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Add a new `used_by_otel` field in the modules defined in `tasks/modules.py`, and use that field to determine whether to update the version of Go in `go.mod` files.

### Motivation

Modules used by OTel need to stay compatible with the previous minor version of Go.

### Additional Notes

The list of modules used in the OTel exporter can be found in their `go.mod` file https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/go.mod.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/update_go.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def go_version(_):
         "test_version": "Whether the image is a test image or not",
         "warn": "Don't exit in case of matching error, just warn.",
         "release_note": "Whether to create a release note or not. The default behaviour is to create a release note",
+        "all_modules": "Whether to update the version in go.mod files used by otel.",
```
Comment (KevinFairise2) [inline tasks/update_go.py:27]: I find the name not very explicit. What about a name mentioning otel, like `include_otel` so we know what this parameter does

**Review by KevinFairise2 — APPROVED** [review]

File: tasks/update_go.py (hunk 1 above)
Comment (GustavoCaso) [reply tasks/update_go.py:27]: Agree 100% 😄 Maybe `include_otel_modules`

**Review by GustavoCaso — APPROVED** [review]
LGTM. I agree with @KevinFairise2 that the task flag name `all_module` could be improve to better describe what the flag actually does

File: tasks/update_go.py (hunk 1 above)
Comment (pgimalac) [reply tasks/update_go.py:27]: I was thinking that we might have to not update some modules other than those used by otel, hence the name, but I guess there is no need to plan for something which might not happen :smile:

**Pull request merged**